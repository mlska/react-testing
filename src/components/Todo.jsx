import React from "react";

export const Todo = ({ todo }) => {
  const { id, desc, completed } = todo;
  return (
    <div
      style={{
        display: "flex",
        justifyContent: "space-between",
        padding: "1rem 5rem",
      }}
      data-testid={`todo-${id}`}
    >
      <div>
        {completed ? (
          <strike data-id={id}>{desc}</strike>
        ) : (
          <span data-id={id}>{desc}</span>
        )}
      </div>
      <button>X</button>
    </div>
  );
};
