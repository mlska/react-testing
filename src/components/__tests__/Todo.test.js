import { render, screen, cleanup } from "@testing-library/react";
import { Todo } from "../Todo";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

test(`should render ToDo non-completed component`, () => {
  const todo = { id: 1, desc: "wash dishes", completed: false };
  render(<Todo todo={todo} />);
  const todoElement = screen.getByTestId("todo-1");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("wash dishes");
  expect(todoElement).not.toContainHTML("<strike>");
});

test(`should render ToDo completed component`, () => {
  const todo = { id: 2, desc: "wash car", completed: true };
  render(<Todo todo={todo} />);
  const todoElement = screen.getByTestId("todo-2");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("wash car");
});

test("matches snapshot", () => {
  const todo = { id: 1, desc: "wash dishes", completed: false };
  const tree = renderer.create(<Todo todo={todo} />).toJSON();
  expect(tree).toMatchSnapshot();
});
