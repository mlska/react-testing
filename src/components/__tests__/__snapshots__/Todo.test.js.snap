// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="todo-1"
  style={
    Object {
      "display": "flex",
      "justifyContent": "space-between",
      "padding": "1rem 5rem",
    }
  }
>
  <div>
    <span
      data-id={1}
    >
      wash dishes
    </span>
  </div>
  <button>
    X
  </button>
</div>
`;
